editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -3191,16 +3191,39 @@ void SpatialEditorViewport::_init_gizmo_instance(int p_idx) {
 
 void SpatialEditorViewport::_finish_gizmo_instances() {
 	for (int i = 0; i < 3; i++) {
-		VS::get_singleton()->free(move_gizmo_instance[i]);
-		VS::get_singleton()->free(move_plane_gizmo_instance[i]);
-		VS::get_singleton()->free(rotate_gizmo_instance[i]);
-		VS::get_singleton()->free(scale_gizmo_instance[i]);
-		VS::get_singleton()->free(scale_plane_gizmo_instance[i]);
+		if (move_gizmo_instance[i].is_valid()) {
+			VS::get_singleton()->free(move_gizmo_instance[i]);
+		}
+		move_gizmo_instance[i] = RID();
+
```

Comment:
As this PR has grown again, it's probably more important to do this a bit more sensibly. These setting of the values to RID() only need to occur inside the if statement:
```
		if (move_gizmo_instance[i].is_valid()) {
			VS::get_singleton()->free(move_gizmo_instance[i]);
		        move_gizmo_instance[i] = RID();
		}
```
This applies to all these cases. Aside from that it looks fine to me, I don't know if @akien-mga is okay with the commit message, he is expert in that area. :slightly_smiling_face: 

It's also worth using "may fix" rather than "fix" (if you are not sure via a reproduction project). And also sometimes it can be good to write the fix aspects outside the commit message itself, just in case we later find it doesn't fix one of the issues. Depends how certain you are it fixes them I guess! :smile: